config: interpret leading ? in config-file as an optional file

gpanders opened the pull request:
For example, `config-file = ?config.foo` will not report an error if `config.foo` does not exist.

mitchellh reviewed (changes requested):
Big thinker comment.

gpanders committed:
config: interpret leading ? in config-file as an optional file

For example, `config-file = ?config.foo` will not report an error if
`config.foo` does not exist.
